coderd/coderdtest/authorize.go

```diff
@@ -288,11 +292,12 @@ func caller(skip int) string {
 }
 
 func (r *RecordingAuthorizer) Authorize(ctx context.Context, subject rbac.Subject, action policy.Action, object rbac.Object) error {
-	r.recordAuthorize(subject, action, object)
 	if r.Wrapped == nil {
 		panic("Developer error: RecordingAuthorizer.Wrapped is nil")
 	}
-	return r.Wrapped.Authorize(ctx, subject, action, object)
+	err := r.Wrapped.Authorize(ctx, subject, action, object)
+	r.recordAuthorize(subject, action, object, err)
```

Comment:
...why doesn't `recordAuthorize` just call `Wrapped.Authorize`? passing in an `error` as an argument is definitely not something I've seen before, so it raises some flags for me

Reply:
You can't really call `Authorize` from `recordAuthorize` because you have to call `Authorize` from the normal `Authorizer`. And from the `PreparedRecorder` (which has 2 behaviors if it's SQL or inmem).

The goal of passing in `err` is to save if the authorize called passed or not. I'll rename the err, how about:

```golang
authzErr := r.Wrapped.Authorize(ctx, subject, action, object)
r.recordAuthorize(subject, action, object, authzErr)
return authzErr
```

Reply:
I think the `recordAuthorize` definition would benefit the most from a better name for the argument it takes, but yeah that'd be nice

Reply:
I renamed it to `authzErr`.